textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/utils/Utils.java

```diff
@@ -130,8 +143,10 @@ public static Schema createSpanSchema(Schema schema) {
         try{
             tokenStream.reset();
             while (tokenStream.incrementToken()) {
-                String term = charTermAttribute.toString();
-                resultSet.add(term);
+                String token = charTermAttribute.toString();
+                int tokenIndex = query.toLowerCase().indexOf(token);

```

Comment:
Why do we do this extra `substring()` step?  Add comments here and to the codebase.